src/tools/concurrency/TraceParser.java

```diff
@@ -0,0 +1,342 @@
+package tools.concurrency;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+import som.VM;
+import som.vm.Symbols;
+import som.vm.VmSettings;
+import som.vmobjects.SSymbol;
+
+
+
+public final class TraceParser {
+  private static final byte ACTOR_CREATION = 1;
+  private static final byte PROMISE_CREATION = 2;
+  private static final byte PROMISE_RESOLUTION = 3;
+  private static final byte PROMISE_CHAINED = 4;
+  private static final byte MAILBOX = 5;
+  private static final byte THREAD = 6;
+  private static final byte MAILBOX_CONTD = 7;
+
+  private static final byte MESSAGE_BIT = (byte) 0x80;
+  private static final byte PROMISE_BIT = 0x40;
+  private static final byte TIMESTAMP_BIT = 0x20;
+  private static final byte PARAMETER_BIT = 0x10;
+
+
+  private final HashMap<Short, SSymbol> symbolMapping = new HashMap<>();
+  private ByteBuffer b = ByteBuffer.allocate(ActorExecutionTrace.BUFFER_SIZE);
+  private final HashMap<Long, ActorNode> mappedActors = new HashMap<>();
+  private final HashMap<Long, Queue<Message>> expectedMessages = new HashMap<>();
+  private long currentReceiver;
+  private long currentMessage;
+  private int currentMailbox;
+  private int msgNo;
+  private long parsedMessages = 0;
+  private long parsedActors = 0;
+
+  private static TraceParser parser;
+
+  public static synchronized Queue<Message> getExpectedMessages(final long replayId) {
+    if (parser == null) {
+      parser = new TraceParser();
+      parser.parseTrace();
+    }
+    return parser.expectedMessages.remove(replayId);
+  }
+
+  public static synchronized long getReplayId(final long parentId, final int childNo) {
+    if (parser == null) {
+      parser = new TraceParser();
+      parser.parseTrace();
+    }
+    assert parser.mappedActors.containsKey(parentId) : "Parent doesn't exist";
+    return parser.mappedActors.get(parentId).getChild(childNo).actorId;
+  }
+
+  private TraceParser() {
+    assert VmSettings.REPLAY;
+  }
+
+  private void parseTrace() {
+    File f = new File(VmSettings.TRACE_FILE + ".trace");
```

Comment:
could those two names please be a bit more explicit? Don't like such 1 or 2 letter names for important things.